mmpose/datasets/datasets/top_down/topdown_posetrack18_video_dataset.py

```diff
@@ -0,0 +1,542 @@
+                continue
+
+            box = det_res['bbox']
+
+            # deal with different bbox file formats
+            if 'nframes' in det_res and 'frame_id' in det_res:
+                nframes = int(det_res['nframes'])
+                frame_id = int(det_res['frame_id'])
+            elif 'image_name' in det_res:
+                img_id = self.name2id[det_res['image_name']]
+                img_ann = self.coco.loadImgs(img_id)[0]
+                nframes = int(img_ann['nframes'])
+                frame_id = int(img_ann['frame_id'])
+            else:
+                img_id = det_res['image_id']
+                img_ann = self.coco.loadImgs(img_id)[0]
+                nframes = int(img_ann['nframes'])
+                frame_id = int(img_ann['frame_id'])
+
+            image_files = []
+            if 'image_name' in det_res:
+                file_name = det_res['image_name']
+            else:
+                file_name = self.id2name[det_res['image_id']]
+
+            cur_image_file = os.path.join(self.img_prefix, file_name)
+            image_files.append(cur_image_file)
+
+            prev_nm = file_name.split('/')[-1]
+            ref_idx = int(prev_nm.replace('.jpg', ''))
+
+            indexes = self.frame_indexes_test
+            for index in indexes:
+                if self.test_mode and index == 0:
+                    continue
+                sup_idx = ref_idx + index
+                sup_idx = np.clip(sup_idx, 0, nframes - 1)
+                sup_image_file = cur_image_file.replace(
+                    prev_nm,
+                    str(sup_idx).zfill(self.ph_fill_len) + '.jpg')
+
+                if os.path.exists(sup_image_file):
+                    image_files.append(sup_image_file)
+                else:
+                    warnings.warn(f'{sup_image_file} does not exist, '
+                                  f'use {cur_image_file} instead.')
+                    image_files.append(cur_image_file)
+
+            center, scale = self._xywh2cs(*box[:4])
+            joints_3d = np.zeros((num_joints, 3), dtype=np.float32)
+            joints_3d_visible = np.ones((num_joints, 3), dtype=np.float32)
+            kpt_db.append({
+                'image_file': image_files,
+                'center': center,
+                'scale': scale,
+                'rotation': 0,
+                'bbox': box[:4],
+                'bbox_score': score,
+                'dataset': self.dataset_name,
+                'joints_3d': joints_3d,
+                'joints_3d_visible': joints_3d_visible,
+                'bbox_id': bbox_id,
+                'nframes': nframes,
+                'frame_id': frame_id,
+                'frame_weight': self.frame_weight
+            })
+            bbox_id = bbox_id + 1
+        print(f'=> Total boxes after filter '
+              f'low score@{self.det_bbox_thr}: {bbox_id}')
+        return kpt_db
+
+    def evaluate(self, outputs, res_folder, metric='mAP', **kwargs):
```

Comment:
This part is not covered by unit-test

Reply:
I will add the corresponding unittest.

Reply:
It seems that when testing the `evaluate` function in both `TopDownPoseTrack18Dataset` and `TopDownPoseTrack18VideoDataset`, we need the corresponding ground truth annotations in folder `tests/data/posetrack/annotations`, which is currently missing.